resources/custom-resources/start-codebuild/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/StartCodeBuild.java

```diff
@@ -0,0 +1,101 @@
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazon.aws.partners.saasfactory.saasboost;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.codebuild.CodeBuildClient;
+import software.amazon.awssdk.services.codebuild.model.Build;
+import software.amazon.awssdk.services.codebuild.model.CodeBuildException;
+import software.amazon.awssdk.services.codebuild.model.StartBuildResponse;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+public class StartCodeBuild implements RequestHandler<Map<String, Object>, Object> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StartCodeBuild.class);
+    private static final String AWS_REGION = System.getenv("AWS_REGION");
+    private final CodeBuildClient codeBuild;
+
+    public StartCodeBuild() {
+        if (Utils.isBlank(AWS_REGION)) {
+            throw new IllegalStateException("Missing required environment variable AWS_REGION");
+        }
+        LOGGER.info("Version Info: {}", Utils.version(this.getClass()));
+        this.codeBuild = Utils.sdkClient(CodeBuildClient.builder(), CodeBuildClient.SERVICE_NAME);
+    }
+
+    @Override
+    public Object handleRequest(Map<String, Object> event, Context context) {
+        Utils.logRequestEvent(event);
+
+        final String requestType = (String) event.get("RequestType");
+        final Map<String, Object> resourceProperties = (Map<String, Object>) event.get("ResourceProperties");
+        final String project = (String) resourceProperties.get("Project");
+
+        ExecutorService service = Executors.newSingleThreadExecutor();
+        Map<String, Object> responseData = new HashMap<>();
+        try {
+            Runnable r = () -> {
+                if ("Create".equalsIgnoreCase(requestType) || "Update".equalsIgnoreCase(requestType)) {
+                    LOGGER.info("CREATE or UPDATE");
+                    try {
+                        StartBuildResponse response = codeBuild.startBuild(request -> request
+                                .projectName(project)
+                                .build()
+                        );
+                        //LOGGER.info(Utils.toJson(response));
+                        Build build = response.getValueForField("build", Build.class).get();
+                        responseData.put("BuildStatus", build.buildStatusAsString());
```

Comment:
I'm not sure we know that _immediately_ after invoking start build. I think it will always be `IN_PROGRESS` (followed by `FAILED` or `FAULT` later in time should something go wrong). I have been thinking about adding a CloudFormation wait condition callback to the `post_build` phase of the buildspec so the stack will know when the build finishes (if that's helpful).